waffle-chai/src/matchers/changeEtherBalance.ts

```diff
@@ -0,0 +1,97 @@
+import {BigNumber, BigNumberish, providers} from 'ethers';
+import {ensure} from './calledOnContract/utils';
+import {Account, getAddressOf} from './misc/account';
+
+export function supportChangeEtherBalance(Assertion: Chai.AssertionStatic) {
+  Assertion.addMethod('changeEtherBalance', function (
+    this: any,
+    account: Account,
+    balanceChange: BigNumberish,
+    options: any
+  ) {
+    const subject = this._obj;
+    const derivedPromise = Promise.all([
+      getBalanceChange(subject, account, options),
+      getAddressOf(account)
+    ]).then(
+      ([actualChange, address]) => {
+        this.assert(
+          actualChange.eq(BigNumber.from(balanceChange)),
+          `Expected "${address}" to change balance by ${balanceChange} wei, ` +
+          `but it has changed by ${actualChange} wei`,
+          `Expected "${address}" to not change balance by ${balanceChange} wei,`,
+          balanceChange,
+          actualChange
+        );
+      }
+    );
+    this.then = derivedPromise.then.bind(derivedPromise);
+    this.catch = derivedPromise.catch.bind(derivedPromise);
+    this.promise = derivedPromise;
+    return this;
+  });
+}
+
+async function getBalanceChange(
+  transaction:
+  | providers.TransactionResponse
+  | (() => Promise<providers.TransactionResponse>
+  | providers.TransactionResponse),
+  account: Account,
+  options: any
+) {
+  if (typeof transaction === 'function') {
+    return getBalanceChangeForTransactionCall(transaction, account, options);
+  } else {
+    return getBalanceChangeForTransactionResponse(transaction, account, options);
+  }
```

Comment:
I know it was like this before, but why not just use `getBalanceChangeForTransactionResponse` for both cases and just `await transaction()` if it's a function? 

This will simplify the code and remove the duplication of L64-L72. Note that a change in docs will be needed as well, because inner working of this matcher are described there.